Bottom: d89d2d7219b857ee6c47ef9ce4de925eec035f6a
Top:    ec3e8628df2417150a394a05153b0f8321bbd354
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-11-11 16:53:31 +0000

We can just delete kicked out holes

Since we don't keep holes from broken contours, if a contour is
kicked out, we know it is destined for the bin.

---

diff --git a/src/polygon1.c b/src/polygon1.c
index d103674..48042a8 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -1315,6 +1315,7 @@ find_inside (const BoxType * b, void *cl)
     {
       return 0;
     }
+
   if (poly_ContourInContour (info->want_inside, check))
     {
       info->result = check;
@@ -1424,7 +1425,8 @@ InsertHoles (jmp_buf * e, POLYAREA * dest, PLINE ** src)
 	}
       else
 	{
-	  /* Need to check if this new hole means we need to kick out any old ones for reprocessing */
+	  /* Need to check if this new hole means we need to kick out any old
+	   * ones for deletion */
 	  while (1)
 	    {
 	      struct find_inside_info info;
@@ -1458,10 +1460,7 @@ InsertHoles (jmp_buf * e, POLYAREA * dest, PLINE ** src)
 
 	      /* Remove hole from the contour */
 	      remove_contour (pa_info->pa, prev, info.result, TRUE);
-
-	      /* Add hole as the next on the list to be processed in this very function */
-	      info.result->next = *src;
-	      *src = info.result;
+	      poly_DelContour (&info.result);
 	    }
 	  /* End check for kicked out holes */
